TeachingRecordSystem/src/TeachingRecordSystem.Api/V3/Core/Operations/SetQTLS.cs

```diff
@@ -0,0 +1,100 @@
+using Microsoft.Xrm.Sdk.Query;
+using TeachingRecordSystem.Api.V3.Core.SharedModels;
+using TeachingRecordSystem.Api.Validation;
+using TeachingRecordSystem.Core.Dqt;
+using TeachingRecordSystem.Core.Dqt.Models;
+using TeachingRecordSystem.Core.Dqt.Queries;
+
+namespace TeachingRecordSystem.Api.V3.Core.Operations;
+
+
+public record SetQTLSCommand(string Trn, DateOnly? QTSDate);
+
+public class SetQTLSHandler(ICrmQueryDispatcher _crmQueryDispatcher)
+{
+    public async Task<QTLSInfo?> Handle(SetQTLSCommand command)
+    {
+        var contact = (await _crmQueryDispatcher.ExecuteQuery(
+            new GetActiveContactByTrnQuery(
+                command.Trn,
+                new ColumnSet(
+                    Contact.Fields.dfeta_TRN,
+                    Contact.Fields.dfeta_InductionStatus,
+                    Contact.Fields.dfeta_qtlsdate,
+                    Contact.Fields.dfeta_QTSDate
+                    )
+                )
+            ))!;
+
+        if (contact == null)
+        {
+            return null;
+        }
+
+        var (induction, hasABInduction) = await GetInductionWithAppropriateBody(contact.Id);
+        if (!CanSetQTLSDate(hasQTS: contact.dfeta_QTSDate.HasValue, overallInductionStatus: contact.dfeta_InductionStatus, inductionStatus: induction?.dfeta_InductionStatus, hasInductionWithAB: hasABInduction, existingQtlsdate: contact.dfeta_qtlsdate, incomingQtlsDate: command.QTSDate))
+        {
+            var _ = await _crmQueryDispatcher.ExecuteQuery(
+                new CreateReviewTaskQuery()
+                {
+                    TeacherId = contact.Id,
+                    Category = "Unable to set QTLSDate",
+                    Description = $"Unable to set QTLSDate {command.QTSDate}",
```

Comment:
Might be worth formatting the date here
```suggestion
                    Description = $"Unable to set QTLSDate {command.QTSDate:dd/MM/yyyy}",
```